src/services/import/zip.ts

```diff
@@ -1,43 +1,45 @@
-const yauzl = require("yauzl");
-const htmlSanitizer = require('../html_sanitizer');
-const becca = require('../../becca/becca');
-const BAttachment = require('../../becca/entities/battachment');
-const markdownService = require('./markdown.js');
-
-/**
- * @param {TaskContext} taskContext
- * @param {Buffer} fileBuffer
- * @param {BNote} importRootNote
- * @returns {Promise<BNote>}
- */
-async function importZip(taskContext, fileBuffer, importRootNote) {
-    /** @type {Object.<string, string>} maps from original noteId (in ZIP file) to newly generated noteId */
-    const noteIdMap = {};
-    /** @type {Object.<string, string>} maps from original attachmentId (in ZIP file) to newly generated attachmentId */
-    const attachmentIdMap = {};
-    const attributes = [];
+import BAttribute = require('../../becca/entities/battribute');
+import utils = require('../../services/utils');
+import log = require('../../services/log');
+import noteService = require('../../services/notes');
+import attributeService = require('../../services/attributes');
+import BBranch = require('../../becca/entities/bbranch');
+import path = require('path');
+import protectedSessionService = require('../protected_session');
+import mimeService = require('./mime');
+import treeService = require('../tree');
+import yauzl = require("yauzl");
+import htmlSanitizer = require('../html_sanitizer');
+import becca = require('../../becca/becca');
+import BAttachment = require('../../becca/entities/battachment');
+import markdownService = require('./markdown');
+import TaskContext = require('../task_context');
+import BNote = require('../../becca/entities/bnote');
+import NoteMeta = require('../meta/note_meta');
+import AttributeMeta = require('../meta/attribute_meta');
+import { Stream } from 'stream';
+import { NoteType } from '../../becca/entities/rows';
+
+interface MetaFile {
+    files: NoteMeta[]
+}
+
+async function importZip(taskContext: TaskContext, fileBuffer: Buffer, importRootNote: BNote): Promise<BNote> {
+    /** maps from original noteId (in ZIP file) to newly generated noteId */
+    const noteIdMap: Record<string, string> = {};
+    /** type maps from original attachmentId (in ZIP file) to newly generated attachmentId */
+    const attachmentIdMap: Record<string, string> = {};
+    const attributes: AttributeMeta[] = [];
     // path => noteId, used only when meta file is not available
-    /** @type {Object.<string, string>} path => noteId | attachmentId */
-    const createdPaths = { '/': importRootNote.noteId, '\\': importRootNote.noteId };
-    let metaFile = null;
-    /** @type {BNote} */
-    let firstNote = null;
-    /** @type {Set.<string>} */
-    const createdNoteIds = new Set();
-
-    function getNewNoteId(origNoteId) {
+    /** path => noteId | attachmentId */
+    const createdPaths: Record<string, string> = { '/': importRootNote.noteId, '\\': importRootNote.noteId };
+    let metaFile!: MetaFile;
+    let firstNote!: BNote;
```

Comment:
Do we gain any value using the definite assignment operator here, vs the type being null or MetaFile?

IE:
```suggestion
    let metaFile!: MetaFile | null = null;
    let firstNote!: BNote | null = null;
```

Reply:
Seems to work with an extra null check at the end of the `importZip` method.